spec/ios_get_app_version_spec.rb

```diff
@@ -5,37 +5,54 @@
     it 'parses the xcconfig file format correctly and gets the public version' do
       xcconfig_mock_content = <<~CONTENT
         // a comment
-        VERSION_SHORT=6
-        VERSION_LONG=6.30.0
+        VERSION_SHORT = 6
+        VERSION_LONG = 6.30.0
       CONTENT
 
-      allow(File).to receive(:exist?).and_return(true)
-
       expect_version(xcconfig_mock_content: xcconfig_mock_content, expected_version: '6.30')
     end
 
     it 'parses the xcconfig file format correctly and gets the public hotfix version' do
       xcconfig_mock_content = <<~CONTENT
-        VERSION_SHORT=6
+        VERSION_SHORT = 6
         // a comment
-        VERSION_LONG=6.30.1
+        VERSION_LONG = 6.30.1
       CONTENT
 
-      allow(File).to receive(:exist?).and_return(true)
-
       expect_version(xcconfig_mock_content: xcconfig_mock_content, expected_version: '6.30.1')
     end
 
-    def expect_version(xcconfig_mock_content:, expected_version:)
-      xcconfig_mock_file_path = File.join('mock', 'file', 'path')
+    it 'throws an error when the file is not found' do
+      file_path = 'file/not/found'
+
+      expect do
+        run_described_fastlane_action(
+          public_version_xcconfig_file: file_path
+        )
+      end.to raise_error(FastlaneCore::Interface::FastlaneError)
+    end
 
-      allow(File).to receive(:open).with(xcconfig_mock_file_path, 'r').and_yield(StringIO.new(xcconfig_mock_content))
+    it "throws an error when there isn't a version configured in the .xcconfig file" do
+      xcconfig_mock_content = <<~CONTENT
+        VERSION_SHORT = 6
+        // a comment
+      CONTENT
+
+      expect do
+        expect_version(xcconfig_mock_content: xcconfig_mock_content, expected_version: 'n/a')
+      end.to raise_error(FastlaneCore::Interface::FastlaneError)
+    end
+
+    def expect_version(xcconfig_mock_content:, expected_version:)
+      allow(File).to receive(:exist?).and_return(true)
```

Comment:
Nitpick. Written like this, the `allow` seems too "generous". It will pass for any input, which I think could backfire on us.

Also, do we even need it now that 

1. There is a dedicated test for "throws an error when the file is not found" which doesn't use this helper
2. This helper creates a file, making the stubbing of the `File` call redundant

?

Reply:
Oh indeed, you're right on the too lenient `allow`, it should have had the parameter... but as you point out, as a file is being created and passed, this isn't necessary anymore. Will change it, nice catch!

Reply:
`allow` removed here: https://github.com/iangmaia/release-toolkit/pull/1/commits/f64166736ab1db4b5f798ae16b531a0159205d3b